cmd/manager/main.go

```diff
@@ -43,27 +43,31 @@ func init() {
 }
 
 func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	var probeAddr string
-	var secureMetrics bool
-	var enableHTTP2 bool
-	var macPrefixesFile string
-	var insecure bool
-	var managerNamespace string
-	var probeImage string
-	var probeOSImage string
-	var registryPort int
-	var registryProtocol string
-	var registryURL string
-	var registryResyncInterval time.Duration
-	var webhookPort int
-	var enforceFirstBoot bool
-	var enforcePowerOff bool
-	var serverResyncInterval time.Duration
-	var powerPollingInterval time.Duration
-	var powerPollingTimeout time.Duration
+	var (
+		metricsAddr            string
+		enableLeaderElection   bool
+		probeAddr              string
+		secureMetrics          bool
+		enableHTTP2            bool
+		macPrefixesFile        string
+		insecure               bool
+		managerNamespace       string
+		probeImage             string
+		probeOSImage           string
+		registryPort           int
+		registryProtocol       string
+		registryURL            string
+		registryResyncInterval time.Duration
+		webhookPort            int
+		enforceFirstBoot       bool
+		enforcePowerOff        bool
+		serverResyncInterval   time.Duration
+		powerPollingInterval   time.Duration
+		powerPollingTimeout    time.Duration
+		discoveryTimeout       time.Duration
+	)
 
+	flag.DurationVar(&discoveryTimeout, "discovery-timeout", 30*time.Minute, "Timeout for BMC discovery")
```

Comment:
As we are booting the `Server` into a discovery mode using the `metalprobe` agent I am not sure we should call the help hin there "BMC discovery". Maybe something like: "Timeout for discovery boot". would fit better.